### PR TITLE
[lldb] Require paused process and frame for "register info" command

### PR DESCRIPTION
**Pull request by DavidSpickett**
Prior to this the command would simply crash when run on a running process.

Of the three register commands, "info" was the only one missing these requirements. On some level it makes sense because you're not going to read a value or modify anything, but practically I think lldb assumes any time you're going to access register related stuff, the process should be paused.

I noticed this debugging with a remote gdb stub, so I've recreated that scenario using attach in a new test case.

### CODE REVIEW

File: lldb/test/API/commands/register/register/register_command/TestRegisters.py [hunk 1]
```diff
@@ -659,3 +659,16 @@ def test_fs_gs_base(self):
...
             "fs_base does not equal to pthread_self() value.",
         )
+
+    def test_process_must_be_stopped(self):
+        """Check that all register commands error when the process is not stopped."""
+        self.build()
+        exe = self.getBuildArtifact("a.out")
+        pid = self.spawnSubprocess(exe, ["wait_for_attach"]).pid
+        self.setAsync(True)
+        self.runCmd("process attach --continue -p %d" % pid)
```
Comment (walter-erquinigo) [inline lldb/test/API/commands/register/register/register_command/TestRegisters.py:669]: shouldn't these lines be swapped? Other than that, this LGTM

**Review by walter-erquinigo — APPROVED** [review]

File: lldb/test/API/commands/register/register/register_command/TestRegisters.py (hunk 1 above)
Comment (DavidSpickett) [reply lldb/test/API/commands/register/register/register_command/TestRegisters.py:669]: What I found was that `process attach --continue` would never return if you weren't in async mode. Whether that's a bug in itself I'm not sure.

Comment (walter-erquinigo) [reply lldb/test/API/commands/register/register/register_command/TestRegisters.py:669]: All good then!

**Comment by jimingham** [comment]
process attach --continue

won't return till the process stops when in sync mode.  That's really what sync mode means: "commands that continue the target don't return till the target stops".

But if the process is hitting a breakpoint or exiting and we aren't exiting the `process attach --continue` command in reaction to that, then that is a bug.

Jim

**Comment by DavidSpickett** [comment]
Thanks for the explanation. Now I understand what I'm seeing is the expected behaviour, and async mode is the expected way to achieve what the test needs.

**Pull request merged**